js/PulsarUIComponent.js

```diff
@@ -35,10 +35,77 @@ PulsarUIComponent.prototype.init = function () {
     this.initCountdown();
 };
 
+PulsarUIComponent.prototype.getDatatableOptions = function ($table) {
+    let dom = '<"dataTables_top"Birf><"dataTables_actions"T>t<"dataTables_bottom"lp>',
+        langEmptyTable = 'There are currently no items to display',
+        pageLength = 25,
+        select = {
+            className: 'dt-row-selected',
+            style: 'multi',
+            selector: 'td.table-selection'
+        };
+    
+    if ($table.length && $table.data('empty-table')) {
+        langEmptyTable = $table.data('empty-table');
+    }
+
+    if ($table.length && $table.data('page-length')) {
+        pageLength = $table.data('page-length');
+    }
+
+    if ($table.length && $table.data('select') === false) {
+        dom = '<"dataTables_top"irf><"dataTables_actions"T><"dt-disable-selection"t><"dataTables_bottom"p>';
+        select = false;
+    }
+
+    const options = {
+        aaSorting: [],
+        bAutoWidth: false,
+        buttons: [],
+        className: 'dt-row-selected',
+        columnDefs: [
+            { 
+                className: 'control', 
+                orderable: false, 
+                targets: 0
+            },
+            { 
+                searchable: false, 
+                targets: [0]
+            },
+            {
+                orderable: false,
+                targets: [0, 1]
+            }
+        ],
+        dom: dom,
+        language: {
+            emptyTable: langEmptyTable,
+            info: "Showing _START_ to _END_ of _TOTAL_ items",
+            infoEmpty: 'No items',
+            infoFiltered: " (filtered from _MAX_ items)",
+            zeroRecords: "No items matched your filter, please clear it and try again",
+            search: "Filter records"
+        },
+        lengthChange: false,
+        pageLength: pageLength,
+        responsive: {
+            details: {
+                type: 'column'
+            }
+        },
+        select: select,
+        selector: 'td.table-selection',
+        stateSave: false,
+        style: 'multi'
+    };
+
+    return options;
+};
+
```

Comment:
This method looks as though it could be static. This could potentially be useful in environments where a developer does not want to instantiate the PulsarUIComponent but _does_ want the Pulsar datatable options.

It should be as simple as changing  ```PulsarUIComponent.prototype.getDatatableOptions = function ($table) {``` to ```PulsarUIComponent.getDatatableOptions = function ($table) {```

We'd then need to change any references to this method to ```PulsarUIComponent.getDatatableOptions(...)```

Reply:
Done.